tests/unit/lms/services/canvas_api/authenticated_test.py

```diff
@@ -0,0 +1,159 @@
+from unittest.mock import call, create_autospec, sentinel
+
+import pytest
+from h_matchers import Any
+
+from lms.services import CanvasAPIAccessTokenError
+from lms.services.canvas_api import BasicClient, TokenStore
+from lms.services.canvas_api.authenticated import CanvasTokenResponseSchema
+from lms.validation import ValidationError
+from tests import factories
+
+
+class TestCanvasTokenResponseSchema:
+    @pytest.mark.parametrize("value", (-1, 0))
+    def test_expires_in_must_be_positive(self, value):
+        response = factories.requests.Response(
+            status_code=200, json_data={"access_token": "required", "expires_in": value}
+        )
+
+        with pytest.raises(ValidationError):
+            CanvasTokenResponseSchema(response).parse()
+
+
+class TestAuthenticatedClient:
+    def test_send(self, authenticated_client, basic_client, oauth_token):
+        result = authenticated_client.send(
+            "METHOD", "/path", sentinel.schema, sentinel.params
+        )
+
+        basic_client.send.assert_called_once_with(
+            "METHOD",
+            "/path",
+            sentinel.schema,
+            sentinel.params,
+            headers={"Authorization": f"Bearer {oauth_token.access_token}"},
+        )
+
+        assert result == basic_client.send.return_value
+
+    def test_send_refreshes_and_retries_for_CanvasAPIAccessTokenError(
+        self, authenticated_client, basic_client, token_response, oauth_token
+    ):
+        basic_client.send.side_effect = (
+            CanvasAPIAccessTokenError,  # The first call should fail
+            token_response,  # Then we expect a refresh call
+            "success",  # Then finally our successful content request
+        )
+
+        call_args = ("METHOD", "/path", sentinel.schema, sentinel.params)
+        result = authenticated_client.send(*call_args)
+
```

Comment:
This isn't very nice because we trigger ourselves to make a call in the middle, but I tried to use the main params to tell the story of: old param, refresh, new param